### ISSUE TITLE
Issue during load xgboost model from xgboost.json

### ISSUE THREAD

**Issue by gawsoftpl** [triton-inference-server/fil_backend#378]
I received error when I want to parse xgboost.json model

docker version: nvidia/tritonserver:23.12-py3

UNAVAILABLE: Unknown: [22:10:17] /rapids_triton/bui |
|                      |         | ld/_deps/treelite-src/src/frontend/xgboost_json.cc: |
|                      |         | 674: Provided JSON could not be parsed as XGBoost m |
|                      |         | odel. Parsing error at offset 737352: Terminate par |
|                      |         | sing due to Handler error.                          |
|                      |         | num_class":"0","num_feature":"37","num_target":"4"},"objective":{"name":"binary:logistic","reg_loss_

**Comment by dbz10** [comment]
Are you using a compatible version of xgboost? https://github.com/triton-inference-server/fil_backend/blob/main/docs/model_support.md#version-compatibility
In particular, it looks like more recent versions of xgboost may not be supported yet. I vaguely remember encountering a similar error when trying to load a model saved with xgboost >= 2.0, though I don't remember if it was exactly the same.

**Comment by susnato** [comment]
How did you solve that issue? @dbz10

**Comment by gawsoftpl** [comment]
I train model with xgboost==1.7.2

**Comment by dbz10** [comment]
@gawsoftpl take a look at the page I linked above. As far as it states, for recent versions of triton the supported xgboost versions are <1.7

Though I'm not sure if that information is completely up to date, perhaps the most recent triton versions support higher xgboost version, but it could be worth a try.

@susnato for my particular issue, it was solved by downgrading xgboost to 1.6.2. But I can't guarantee my issue was the same one as OP...
